feat(icons): added `share-ios` icon

oosawy opened the pull request:
<!-- Thank you for contributing! -->

<!-- Insert `closes #issueNumber` here if merging this PR will resolve an existing issue -->

## What is the purpose of this pull request?
- [x] New Icon

### Description
Added new `share-ios` icon.

### Icon use case
<!-- What is the purpose of this icon? For each icon added, please insert at least two real life use cases (the more the better). Text like "it's a car icon" is not accepted. -->
- The `share` icon but feels more like Apple.
- Instruction indicating the Safari Share icon.
- Enhancing the Platform-Adaptive experience.

### Alternative icon designs
<!-- If you have any alternative icon designs, please attach them here. -->

## Icon Design Checklist

### Concept
<!-- All of these requirements must be fulfilled. -->
- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license
<!-- Please choose one of the following, and put an "x" next to it. -->
- [ ] The icons are solely my own creation.
- [x] The icons were originally created in #<issueNumber> by @colebemis
- [x] I've based them on the following Lucide icons: `share`
- [ ] I've based them on the following design: <!-- provide source URL and license permitting use -->

### Naming
<!-- All of these requirements must be fulfilled. -->
- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design
<!-- All of these requirements must be fulfilled. -->
- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to three points of precision.

## Before Submitting
- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

karsa-mistmere commented:
Hi @oosawy,

I can't really see us moving forward with this as is for the following reasons:
- the top of the square is half a pixel off grid, and would need better pixel perfection
- the name `share-3` goes against our naming conventions, it would have to be named something that makes it unique, but:
- there just isn't enough difference to set this apart from `share`, at most we could consider giving `share` an update in style, but that would require a little "market research" on what most of out users would prefer

oosawy commented:
Thanks for the feedback!

First, I fixed the pixel misalignment since I needed it for my own use as well.

I agree that there isn’t enough difference to justify a separate icon from `share`. And I'm also not sure if updating the style of the existing icon makes sense.

I can’t do an actual market research, but [I looked into other icon libraries](https://react-icons.github.io/react-icons/search/#q=share) for reference. For share icons with a square and an upward arrow, most of them tend to have a closed square. Examples include `IoIosShare`, `IoShare`, `MdIosShare`, `GoShare`, `TbShare2`, and `RxShare2`. On the other hand, some like `RiShare2Line`, `FiShare`, and our `LuShare` are open square.

That said, I still feel like it’s a pretty minor difference. Since my use case (PWA install instruction) is quite specific, I’m not sure if changing the `share` style is necessary. I do like that the standard variant stays neutral and isn’t overly tied to Apple’s design.

To make this PR easier to find, I renamed `share-3` to `share-ios`, based on the naming conventions used in some icon libraries (though I’m not sure if a brand name like this would be acceptable here).

karsa-mistmere commented:
> To make this PR easier to find, I renamed `share-3` to `share-ios`, based on the naming conventions used in some icon libraries (though I’m not sure if a brand name like this would be acceptable here).

iOS is a trademark, we cannot officially use it.
